src/seedu/addressbook/data/AddressBook.java

```diff
@@ -74,4 +78,15 @@ public boolean equals(Object other) {
                 || (other instanceof AddressBook // instanceof handles nulls
                         && this.allPersons.equals(((AddressBook) other).allPersons));
     }
+
+    public List<ReadOnlyPerson> sortPersons( List<ReadOnlyPerson> allPersons) {
+        if (allPersons.size() > 1) {
+            Collections.sort(allPersons, new Comparator<ReadOnlyPerson>() {
+                public int compare(ReadOnlyPerson a, ReadOnlyPerson b) {
+                    return a.getName().compareTo(b.getName());
```

Comment:
You are comparing two Name objects using the compareTo method. However, the comapreTo implementation in Name Class is not available. This seems a compilation error.